From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Tue, 14 May 2024 18:00:07 +0200
Subject: [PATCH] Modify systemd service for Arch

Start after Samba and keep backward compatibility with the AUR package.
---
 etc/systemd/wsdd-discovery.service       | 4 ++--
 etc/systemd/{wsdd.defaults => wsdd.conf} | 2 +-
 etc/systemd/wsdd.service                 | 8 +++-----
 3 files changed, 6 insertions(+), 8 deletions(-)
 rename etc/systemd/{wsdd.defaults => wsdd.conf} (84%)

diff --git a/etc/systemd/wsdd-discovery.service b/etc/systemd/wsdd-discovery.service
index 7f13202c8aa7..2aac5e40d2c6 100644
--- a/etc/systemd/wsdd-discovery.service
+++ b/etc/systemd/wsdd-discovery.service
@@ -5,8 +5,8 @@ Requires=wsdd-discovery.socket
 
 [Service]
 Type=simple
-; Use /etc/default/wsdd for defaults, if it exists
-EnvironmentFile=-/etc/default/wsdd
+; Use /etc/conf.d/wsdd for defaults, if it exists
+EnvironmentFile=-/etc/conf.d/wsdd
 ; The service is put into an empty runtime directory chroot,
 ; i.e. the runtime directory which usually resides under /run
 ExecStart=/usr/bin/wsdd --shortlog --chroot=/run/wsdd-discovery --source-port=37020 --no-host --discovery $WSDD_DISCOVERY_PARAMS
diff --git a/etc/systemd/wsdd.defaults b/etc/systemd/wsdd.conf
similarity index 84%
rename from etc/systemd/wsdd.defaults
rename to etc/systemd/wsdd.conf
index d7172577da26..1b58ac509fba 100644
--- a/etc/systemd/wsdd.defaults
+++ b/etc/systemd/wsdd.conf
@@ -2,5 +2,5 @@
 # Use, e.g., "-i eth0" to restrict operations to a specific interface
 # Refer to the wsdd(8) man page for details
 
-WSDD_PARAMS=""
+WSDD_PARAMS="--workgroup WORKGROUP"
 WSDD_DISCOVERY_PARAMS=""
diff --git a/etc/systemd/wsdd.service b/etc/systemd/wsdd.service
index 6bb951d9a942..d979afd5c090 100644
--- a/etc/systemd/wsdd.service
+++ b/etc/systemd/wsdd.service
@@ -5,15 +5,13 @@ Documentation=man:wsdd(8)
 After=network-online.target
 Wants=network-online.target
 ; It makes sense to have Samba running when wsdd starts, but is not required.
-; Thus, the next to lines are disabled and use BindsTo only.
-; One may also add any of these services to After for stronger binding.
 ;BindsTo=smb.service
-;BindsTo=samba.service
+After=smb.service
 
 [Service]
 Type=simple
-; Use /etc/default/wsdd for defaults, if it exists
-EnvironmentFile=-/etc/default/wsdd
+; Use /etc/conf.d/wsdd for defaults, if it exists
+EnvironmentFile=-/etc/conf.d/wsdd
 ; The service is put into an empty runtime directory chroot,
 ; i.e. the runtime directory which usually resides under /run
 ExecStart=/usr/bin/wsdd --shortlog --chroot=/run/wsdd $WSDD_PARAMS
